back/src/forms/permissions.ts

```diff
@@ -1,55 +1,97 @@
 import { Form, User, TemporaryStorageDetail } from "@prisma/client";
 import { ForbiddenError } from "apollo-server-express";
 import prisma from "../prisma";
-import { FormSirets } from "./types";
+import { FormCompanies } from "./types";
 
-import { getCachedUserSirets } from "../common/redis/users";
+import {
+  getCachedUserCompanies,
+  getCachedUserSirets
+} from "../common/redis/users";
 
 import { getFullForm } from "./database";
 import { InvaliSecurityCode, NotFormContributor } from "./errors";
 import { getFormRepository } from "./repository";
 
-function isFormEmitter(userSirets: string[], form: FormSirets) {
+export async function formToCompanies(form: Form): Promise<FormCompanies> {
+  const fullForm = await getFullForm(form);
+  return {
+    emitterCompanySiret: fullForm.emitterCompanySiret,
+    recipientCompanySiret: fullForm.recipientCompanySiret,
+    transporterCompanySiret: fullForm.transporterCompanySiret,
+    traderCompanySiret: fullForm.traderCompanySiret,
+    brokerCompanySiret: fullForm.brokerCompanySiret,
+    ecoOrganismeSiret: fullForm.ecoOrganismeSiret,
+    ...(fullForm.intermediaries?.length
+      ? {
+          intermediariesVatNumber: fullForm.intermediaries.map(
```

Comment:
Comme pour les transportSegments, vu que c'est des tableaux p-e mettre au pluriel
```suggestion
          intermediariesVatNumbers: fullForm.intermediaries.map(
```